### PR TITLE
Feature/favorite articles

### PR DESCRIPTION
**Pull request by madhaus4**
# Description
Add toggling favoriting functionality to articles.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
- [x] Use console.log()
- [x] Dev tools / react dev tools
- [x] View in local host
- [ ] Verify acceptance criteria (if applicable)

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

#14

### CODE REVIEW

**Comment by madhaus4** [comment]
@sertmer @kaylagordon

File: src/components/App/App.js [hunk 1]
```diff
@@ -10,8 +10,8 @@ import './App.css';
 function App() {
   const [history, setHistory] = useState([])
   const [rockets, setRockets] = useState([])
-  const [error, setError] = useState('')
   const [launchImages, setLaunchImages] = useState([])
+  const [error, setError] = useState('')
```
Comment (Meggs625) [inline src/components/App/App.js:14]: It does feel cleaner at the bottom

File: src/components/HistoricalCards/HistoricalCards.js [hunk 2]
```diff
@@ -1,40 +1,25 @@
...
 import { ReactTinyLink } from 'react-tiny-link';
-// import dayjs from 'dayjs';
-
-function HistoricalCards({ title, image, details, link}) {
+import './HistoricalCards.css';
 
+function HistoricalCards({ story, image, updateFavorites }) {
   return (
-    <article className='historical-card'>
+    <article className='historical-card' id={story.id}>
```
Comment (Meggs625) [inline src/components/HistoricalCards/HistoricalCards.js:6]: Looks solid - great job getting everything to work with the change up

File: src/components/HistoricalContainer/HistoricalContainer.js [hunk 3]
```diff
@@ -1,22 +1,62 @@
...
+        updateFavorites={updateFavorites}
+      /> 
+    ))
+  }
+    
   return(
     <section className='historical-cards-section'>
-      {allHistory}
+      <button 
+        onClick={() => toggleDisplay()}
```
Comment (Meggs625) [inline src/components/HistoricalContainer/HistoricalContainer.js:51]: possible refactor to just have one button whose text changes with the page display

**Review by Meggs625 — APPROVED** [review]
So glad we got this going! Woo hoo!

File: src/components/HistoricalContainer/HistoricalContainer.js [hunk 4]
```diff
@@ -1,22 +1,62 @@
...
-      details={story.details}
-      link={story.links.article} 
-      image={launchImages[i]} 
-    /> 
-  ))
+  const [favoritesList, setFavoritesList] = useState([])
+  const [isFavorited, setIsFavorited] = useState(false)
+
+  const updateFavorites = (story) => {
+    let foundFavorite = favoritesList.find(favorite => favorite.id === story.id)
```
Comment (kaylagordon) [inline src/components/HistoricalContainer/HistoricalContainer.js:10]: great variable naming - this line is easy for me to read, which is great b/c I've never seen this code before!

File: src/components/HistoricalContainer/HistoricalContainer.js [hunk 5]
```diff
@@ -1,22 +1,62 @@
...
+
+    if (foundFavorite) {
+      removeFromFavorites(foundFavorite)
+    } else {
+      addToFavorites(story)
+    }
+  }
   
+  const addToFavorites = (story) => {
```
Comment (kaylagordon) [inline src/components/HistoricalContainer/HistoricalContainer.js:19]: Are you planning to do anything to ensure that these favorites persist on page load? Local storage would be an easy win!

File: src/components/HistoricalContainer/HistoricalContainer.js [hunk 6]
```diff
@@ -1,22 +1,62 @@
...
+    }
+    setFavoritesList([...favoritesList, newFavorite])
+  }
+
+  const removeFromFavorites = (story) => {
+    let itemsToKeep = favoritesList.filter(favorite => favorite.id !== story.id)
+    setFavoritesList(itemsToKeep)
+  }
+
+  const toggleDisplay = () => {
```
Comment (kaylagordon) [inline src/components/HistoricalContainer/HistoricalContainer.js:31]: I don't know if `toggleDisplay` is a great name. Maybe `toggleFavorite` would be more clear?

File: src/components/HistoricalContainer/HistoricalContainer.js [hunk 7]
```diff
@@ -1,22 +1,62 @@
...
-      {allHistory}
+      <button 
+        onClick={() => toggleDisplay()}
+        >View Favorites
+      </button>
+      <button 
+        onClick={() => toggleDisplay()}
+        >View All
+      </button>
+      {!isFavorited && displayArticles(theHistory)}
```
Comment (kaylagordon) [inline src/components/HistoricalContainer/HistoricalContainer.js:58]: I would not include articles like `the` in a variable name

**Pull request merged**